rules/always-return.js

```diff
@@ -124,6 +124,34 @@ function peek(arr) {
   return arr[arr.length - 1]
 }
 
+/**
+ * Checks if the node is an assignment to an ignored variable
+ * @param {Node} node
+ * @param {string[]} ignoredVars
+ * @returns {boolean}
+ */
+function isIgnoredAssignment(node, ignoredVars) {
+  if (node.type !== 'ExpressionStatement') return false
+  const expr = node.expression
+  if (expr.type !== 'AssignmentExpression') return false
+  const left = expr.left
+  // istanbul ignore else
+  if (left.type === 'MemberExpression') {
+    if (left.object.type === 'Identifier') {
+      return ignoredVars.includes(left.object.name)
+    }
+    if (
+      left.object.type === 'MemberExpression' &&
+      left.object.object.type === 'Identifier'
+    ) {
+      const fullName = `${left.object.object.name}.${left.object.property.name}`
```

Comment:
**thought:** I wonder if we should also consider things like `foo['bar']` (`left.computed === 
false && left.property.type === 'Literal'`)

I am open to if you think this is worth covering, I just figure i would bring it up :)